x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/JobConfigProviderIT.java

```diff
@@ -68,18 +68,25 @@ public void testGetMissingJob() throws InterruptedException {
 
         assertNull(jobHolder.get());
         assertNotNull(exceptionHolder.get());
-        assertThat(exceptionHolder.get(), instanceOf(ResourceNotFoundException.class));
+        assertEquals(ResourceNotFoundException.class, exceptionHolder.get().getClass());
     }
 
     public void testCheckJobExists() throws InterruptedException {
         AtomicReference<Boolean> jobExistsHolder = new AtomicReference<>();
         AtomicReference<Exception> exceptionHolder = new AtomicReference<>();
 
-        blockingCall(actionListener -> jobConfigProvider.checkJobExists("missing", actionListener), jobExistsHolder, exceptionHolder);
-
-        assertNull(jobExistsHolder.get());
-        assertNotNull(exceptionHolder.get());
-        assertThat(exceptionHolder.get(), instanceOf(ResourceNotFoundException.class));
+        boolean throwIfMissing = true; //randomBoolean();
```

Comment:
Is the `true` a leftover from debugging?

Reply:
Ha yes good catch